src/Apps/Settings/Routes/EditProfile/Components/SettingsEditProfileFields.tsx

```diff
@@ -90,11 +90,16 @@ const SettingsEditProfileFields: React.FC<SettingsEditProfileFieldsProps> = ({
 
   const onSubmit = async (values: EditProfileFormModel) => {
     try {
-      const newLocation = { ...values.location }
-      delete newLocation.display
+      const location = {
+        city: values.location?.city || null,
+        state: values.location?.state || null,
+        country: values.location?.country || null,
+        countryCode: values.location?.countryCode || null,
+      }
```

Comment:
We have to set every value to avoid something like this:

![Screenshot 2024-05-23 at 11 17 54](https://github.com/artsy/force/assets/4691889/346d0180-193e-41b7-ba97-37bbe5c33b59)